pyteal/ast/abi/string.py

```diff
@@ -67,7 +68,7 @@ def set(
         match value:
             case ComputedValue():
                 if value.produced_type_spec() == StringTypeSpec():
-                    return value.store_into(self)
+                    return value.store_into(cast(DynamicArray[T], self))
```

Comment:
nit: I think you can circumvent both cast cases by removing `T` type var in `set` signature, by using `Byte` directly?

Reply:
This worked for removing the cast on computed but not on the sequence because it thinks its possible for `str` and `byte` types to make it there:

```
pyteal/ast/abi/string.py:90: error: Argument 1 to "set" of "DynamicArray" has incompatible type "Union[Sequence[Byte], str, bytes]"; expected "Union[Sequence[Byte], DynamicArray[Byte], ComputedValue[DynamicArray[Byte]]]"
```

Reply:
ok I see, then that makes sense. I guess you just need to remove the `T` type var.